src/shogun/lib/DynamicArray.h

```diff
@@ -405,9 +468,10 @@ template <class T> class CDynamicArray :public CSGObject
 		 *
 		 * @return the array
 		 */
-		inline T* get_array() const
+		inline T* get_array()
 		{
-			return m_array.get_array();
+			auto it = m_array.begin();
+			return &(*it);
```

Comment:
namely i dont think that this can be made work with deque, unless you copy the content from it to a vector temporarily and return that vector's memory region, but then how would you release the memory...